File: model/src/main/java/alive/field/FieldMatrix.java
package alive.field;

import alive.Entity;


public class FieldMatrix implements Field {

    Entity[][] entities;

    public FieldMatrix(int width, int height) {
        entities = new Entity[height][width];
    }

    @Override
    public Entity get(PositionCartesian position) {
        return entities[position.y()][position.x()];
    }

    @Override
    public boolean place(Entity entity, PositionCartesian position) {
        if (!isInBounds(position)) {
            return false;
        }
        entities[position.y()][position.x()] = entity;
        return true;
    }

    @Override
    public boolean relocate(PositionCartesian oldPosition, PositionCartesian newPosition) {
        if (!isInBounds(newPosition)) {
            return false;
        }
        return true;
    }

    @Override
    public boolean isInBounds(PositionCartesian position) {
        return false;
    }
}

File: front/src/main/js/src/components/FieldMatrix.js
import React from 'react'
import Cell from "./Cell";

const FieldMatrix = ({field, height, width}) => (
    <svg viewBox={`0 0 ${width} ${height}`}>
        {
            field.map((xAxis, i) =>
                xAxis.map((entity, j) =>
                    <Cell key={`x${i}y${j}`} size={1} fill={entity.fill} x={i} y={j}/>
                )
            )
        }
    </svg>
)

export default FieldMatrix